### PR TITLE
add ein_60

### PR DESCRIPTION
**Pull request by Klackygears**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Klackygears** [comment]
oops. I need to change the "layout_4x6_2"

**Comment by Klackygears** [comment]
Should be good to go?

File: keyboards/ein_60/config.h [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+// /*== or choose animations ==*/
+//   #define RGBLIGHT_EFFECT_BREATHING
+//   #define RGBLIGHT_EFFECT_RAINBOW_MOOD
+//   #define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+//   #define RGBLIGHT_EFFECT_SNAKE
+//   #define RGBLIGHT_EFFECT_KNIGHT
+//   #define RGBLIGHT_EFFECT_CHRISTMAS
+//   #define RGBLIGHT_EFFECT_STATIC_GRADIENT
+//   #define RGBLIGHT_EFFECT_RGB_TEST
+//   #define RGBLIGHT_EFFECT_ALTERNATING
```
Comment (drashna) [inline keyboards/ein_60/config.h:75]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/ein_60/glcdfont.c [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+#define FONT5X7_H
+
+#ifdef __AVR__
+#    include <avr/io.h>
+#    include <avr/pgmspace.h>
+#elif defined(ESP8266)
+#    include <pgmspace.h>
+#else
+#    define PROGMEM
+#endif
```
Comment (drashna) [inline keyboards/ein_60/glcdfont.c:14]:
```suggestion
#pragma once

#include "progmem.h"
```

File: keyboards/ein_60/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+#include "stdio.h"
```
Comment (drashna) [inline keyboards/ein_60/keymaps/default/keymap.c:17]:
```suggestion
```

File: keyboards/ein_60/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    //oled_write_ln_P(PSTR(""), false);
+    //render_ein60_logo();
+
+    // Host Keyboard LED Status
+    led_t led_state = host_keyboard_led_state();
+    oled_write_P(led_state.num_lock ? PSTR("NUM ") : PSTR("    "), false);
+    oled_write_P(led_state.caps_lock ? PSTR("CAP ") : PSTR("    "), false);
+    oled_write_P(led_state.scroll_lock ? PSTR("SCR ") : PSTR("    "), false);
+
+    sprintf(wpm_str, "WPM: %03d", get_current_wpm());
```
Comment (drashna) [inline keyboards/ein_60/keymaps/default/keymap.c:169]: This uses up ~1.5kb of firmware space.  So, bad. 

```suggestion
    uint8_t n = get_current_wpm();
    char    wpm_counter[4];
    wpm_counter[3] = '\0';
    wpm_counter[2] = '0' + n % 10;
    wpm_counter[1] = (n /= 10) % 10 ? '0' + (n) % 10 : (n / 10) % 10 ? '0' : ' ';
    wpm_counter[0] = n / 10 ? '0' + n / 10 : ' ';
    oled_write_P(PSTR("WPM: "), false);
    oled_write(wpm_counter, false);
```

File: keyboards/ein_60/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+void oled_task_user(void) {
+
+    render_status();
+    oled_write_ln_P(PSTR(""), false);
+    render_ein60_logo();
+}
+#endif
+
+
+void encoder_update_user(uint8_t index, bool clockwise) {
```
Comment (drashna) [inline keyboards/ein_60/keymaps/default/keymap.c:183]:
```suggestion
bool encoder_update_user(uint8_t index, bool clockwise) {
```

File: keyboards/ein_60/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+     KC_CAPS, _______, _______, _______, _______, _______,          _______, _______, _______, _______, _______, _______,
+     _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+    )
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
+#ifdef OLED_DRIVER_ENABLE
```
Comment (drashna) [inline keyboards/ein_60/keymaps/default/keymap.c:124]:
```suggestion
#ifdef OLED_ENABLE
```

Also, ideally, all of the oled and encoder code should be moved to the keyboard's C file.  Eg: 
https://github.com/qmk/qmk_firmware/blob/a2d43ca024d8e840c882cc72e80e9eb6b70384f1/keyboards/splitkb/kyria/rev1/rev1.c#L58-L148

File: keyboards/ein_60/keymaps/default/rules.mk [hunk 7]
```diff
@@ -0,0 +1,5 @@
+# Build Options
+#   change yes to no to disable
+RGBLIGHT_ENABLE = no      # Enable keyboard RGB underglow
+RGB_MATRIX_ENABLE = yes     # Enable for pretty RGB matrix effects
+RGB_MATRIX_DRIVER = WS2812  # Define RGB Matrix
```
Comment (drashna) [inline keyboards/ein_60/keymaps/default/rules.mk:5]: This should go in the keyboard's rules.mk, instead.
```suggestion
```

File: keyboards/ein_60/rules.mk [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+MCU = at90usb1286
+
+# Bootloader selection
+BOOTLOADER = qmk-dfu
+##LAYOUTS = split_4x6_2
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/ein_60/rules.mk:11]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/ein_60/rules.mk [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/ein_60/rules.mk:22]:
```suggestion
```

File: keyboards/ein_60/rules.mk [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+AUDIO_ENABLE = no           # Audio output
+RGB_MATRIX_ENABLE = no       # Enable for pretty RGB matrix effects
+#RGB_MATRIX_DRIVER = WS2812  # Define RGB Matrix
```
Comment (drashna) [inline keyboards/ein_60/rules.mk:25]:
```suggestion
RGB_MATRIX_DRIVER = WS2812  # Define RGB Matrix
```

File: keyboards/ein_60/rules.mk [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+ENCODER_ENABLE = yes        # Enables the use of one or more encoders
+OLED_DRIVER_ENABLE = yes    # Enables the use of OLED displays
```
Comment (drashna) [inline keyboards/ein_60/rules.mk:27]:
```suggestion
OLED_ENABLE = yes    # Enables the use of OLED displays
```

**Review by drashna — CHANGES_REQUESTED** [review]
You have changes to the git submodules that need to be fixed.  This should be fixed by rebasing the branch that was used to open the PR.

**Comment by Klackygears** [comment]
> You have changes to the git submodules that need to be fixed. This should be fixed by rebasing the branch that was used to open the PR.

Shoot! I thought I did that! I’ll work on it!

**Comment by Klackygears** [comment]
> You have changes to the git submodules that need to be fixed. This should be fixed by rebasing the branch that was used to open the PR.

Regarding this, does using the below command not work? git is sometimes really confusing for me.
```
git submodule update --init
```

**Comment by Klackygears** [comment]
Hey @drashna I removed the WPM thing but, I'm wondering if it's a problem? I think there's a lot of space on the AT90USB1286 if that's the concern. However, as always, I may not know anything about what I'm saying lol.

**Comment by Klackygears** [comment]
I added in my keymap with the with the wpm on the oled so that it's available that way. Do you think that's better?

File: keyboards/ein_60/keymaps/klackygears/config.h [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+
+
+#define PERMISSIVE_HOLD
+#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline keyboards/ein_60/keymaps/klackygears/config.h:22]: This is no longer used.
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by drashna** [comment]
> Hey @drashna I removed the WPM thing but, I'm wondering if it's a problem? I think there's a lot of space on the AT90USB1286 if that's the concern. However, as always, I may not know anything about what I'm saying lol.

Fair point.  But it's probably best to just not use the snprintf stuff anyways.

**Comment by Klackygears** [comment]
> > Hey @drashna I removed the WPM thing but, I'm wondering if it's a problem? I think there's a lot of space on the AT90USB1286 if that's the concern. However, as always, I may not know anything about what I'm saying lol.
> 
> Fair point. But it's probably best to just not use the snprintf stuff anyways.

Cool! I'll have to learn more about that :-)

**Review by drashna — APPROVED** [review]

File: keyboards/ein_60/keymaps/default/readme.md [hunk 13]
```diff
@@ -0,0 +1 @@
+# The default keymap for EIN_60. It utilizes a nearly identical layout as the Plank. The OLED is enabled, rotary encoders are enabled, and the LEDs are enabled with RGB Matrix. You'll need to alter the rules.mk file in your own keymap to change this or simply cut the jumper pad with a hobby knife to disconnect them completely. There is an option to install a KELIKING KLJ-1102 speaker and enable with "AUDIO_ENABLE = yes" in your rules.mk file.
```
Comment (fauxpark) [inline keyboards/ein_60/keymaps/default/readme.md:1]:
```suggestion
# The default keymap for EIN_60.

It utilizes a nearly identical layout as the Plank. The OLED is enabled, rotary encoders are enabled, and the LEDs are enabled with RGB Matrix. You'll need to alter the rules.mk file in your own keymap to change this or simply cut the jumper pad with a hobby knife to disconnect them completely. There is an option to install a KELIKING KLJ-1102 speaker and enable with "AUDIO_ENABLE = yes" in your rules.mk file.
```

File: keyboards/ein_60/keymaps/klackygears/readme.md [hunk 14]
```diff
@@ -0,0 +1 @@
+# This is the klackygears userspace keymap for EIN_60. My userspace is heavily based on Drashna's userspace and draws a lot of features from him.
```
Comment (fauxpark) [inline keyboards/ein_60/keymaps/klackygears/readme.md:1]:
```suggestion
This is the klackygears userspace keymap for EIN_60. My userspace is heavily based on Drashna's userspace and draws a lot of features from him.
```

File: keyboards/ein_60/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,208 @@
...
+     * |      |      |      |      |      |      |  |      |  |      |      |  Up  |      |      |      |
+     * |------+------+------+------+------+------|  `------'  |------+------+------+------+------+------|
+     * |      |      |      |      |      |      |            |      | Left | Down |Right |      |      |
+     * |------+------+------+------+------+------|            |------+------+------+------+------+------|
+     * | Caps |      |      |      |      |      |            |      |      |      |      |      |      |
+     * |------+------+------|--------------------+  ,------.  +--------------------|------+------+------|
+     * |      |      |      |      |      |      |  |      |  |      |      |      |      |      |      |
+     * `-----------------------------------------'  `------'  `-----------------------------------------'
+     */
+    [_FUNCTION] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/ein_60/keymaps/default/keymap.c:111]:
```suggestion
    [_FUNCTION] = LAYOUT(
```

File: keyboards/ein_60/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,208 @@
...
+     * |      |      |      |      |      |      |  |      |  |RGBRST|RGBMOD|RGBVAI|RGBSAI|RGBHUI|      |
+     * |------+------+------+------+------+------|  `------'  |------+------+------+------+------+------|
+     * |      |      |      |      |      |      |            |      |RGBRMO|RGBVAD|RGBSAD|RGBHUD|RGBTOG|
+     * |------+------+------+------+------+------|            |------+------+------+------+------+------|
+     * |      |      |      |      |      |      |            |      |      |      |      |      |BLSTEP|
+     * |------+------+------|--------------------+  ,------.  +--------------------|------+------+------|
+     * |      |      |      |      |      |      |  |      |  |      |      |      |      |      | RESET|
+     * `-----------------------------------------'  `------'  `-----------------------------------------'
+     */
+    [_ADJUST] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/ein_60/keymaps/default/keymap.c:94]:
```suggestion
    [_ADJUST] = LAYOUT(
```

File: keyboards/ein_60/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,208 @@
...
+     * | Esc  |   1  |   2  |   3  |   4  |   5  |  |      |  |   6  |   7  |   8  |   9  |   0  | Del  |
+     * |------+------+------+------+------+------|  `------'  |------+------+------+------+------+------|
+     * |      |   4  |   5  |   6  |   +  |  F5  |            |  F6  |   -  |   =  |   [  |   ]  |Enter |
+     * |------+------+------+------+------+------|            |------+------+------+------+------+------|
+     * |Enter |   7  |   8  |   9  |   -  |  F11 |            |  F12 |ISO # |ISO / | Mute |      |   \  |
+     * |------+------+------|--------------------+  ,------.  +--------------------|------+------+------|
+     * |      |   ,  |   0  |  .   |Lower | Bksp |  |      |  | Bksp |Raise | Next | Vol- | Vol+ | Play |
+     * `-----------------------------------------'  `------'  `-----------------------------------------'
+     */
+    [_RAISE] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/ein_60/keymaps/default/keymap.c:77]:
```suggestion
    [_RAISE] = LAYOUT(
```

File: keyboards/ein_60/keymaps/default/keymap.c [hunk 18]
```diff
@@ -0,0 +1,208 @@
...
+     * | Esc  |   !  |   @  |   #  |   $  |   %  |  |      |  |   ^  |   &  |   *  |   (  |   )  | Del  |
+     * |------+------+------+------+------+------|  `------'  |------+------+------+------+------+------|
+     * |  F1  |  F2  |  F3  |  F4  |  F5  |  F6  |            |      |   _  |   +  |   {  |   }  |Enter |
+     * |------+------+------+------+------+------|            |------+------+------+------+------+------|
+     * |  F7  |  F8  |  F9  |  F10 |  F11 |  F12 |            |      |      |      | Mute |      |  |   |
+     * |------+------+------|--------------------+  ,------.  +--------------------|------+------+------|
+     * |      |      |      |      |Lower | Bksp |  |      |  | Bksp |Raise | Next | Vol- | Vol+ | Play |
+     * `-----------------------------------------'  `------'  `-----------------------------------------'
+     */
+    [_LOWER] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/ein_60/keymaps/default/keymap.c:60]:
```suggestion
    [_LOWER] = LAYOUT(
```

File: keyboards/ein_60/keymaps/default/keymap.c [hunk 19]
```diff
@@ -0,0 +1,208 @@
...
+     * | Tab  |   Q  |   W  |   E  |   R  |   T  |  | Mute |  |   Y  |   U  |   I  |   O  |   P  | Bksp |
+     * |------+------+------+------+------+------|  `------'  |------+------+------+------+------+------|
+     * | Func |   A  |   S  |   D  |   F  |   G  |            |   H  |   J  |   K  |   L  |   ;  | Enter|
+     * |------+------+------+------+------+------|            |------+------+------+------+------+------|
+     * | Shift|   Z  |   X  |   C  |   V  |   B  |            |   N  |   M  |   ,  |   .  |   /  |  '   |
+     * |------+------+------|--------------------+  ,------.  +--------------------|------+------+------|
+     * | Ctrl |  `   | GUI  | Alt  |Lower |Space |  |Space |  |Space |Raise | Left | Down |  Up  |Right |
+     * `-----------------------------------------'  `------'  `-----------------------------------------'
+     */
+    [_QWERTY] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/ein_60/keymaps/default/keymap.c:43]:
```suggestion
    [_QWERTY] = LAYOUT(
```

File: keyboards/ein_60/keymaps/ledtest/keymap.c [hunk 20]
```diff
@@ -0,0 +1,208 @@
...
+     * | Tab  |   Q  |   W  |   E  |   R  |   T  |  | Esc  |  |RGBRST|RGBHUI|RGBSAI|RGBVAI|RGBSPI| Bksp |
+     * |------+------+------+------+------+------|  `------'  |------+------+------+------+------+------|
+     * | Func |   A  |   S  |   D  |   F  |   G  |            |RGBRMO|RGBHUD|RGBSAD|RGBVAD|RGBSPD|Enter |
+     * |------+------+------+------+------+------|            |------+------+------+------+------+------|
+     * | Shift|   Z  |   X  |   C  |   V  |   B  |            |RGBTOG|   M  |   ,  |   .  |   /  |  '   |
+     * |------+------+------|--------------------+  ,------.  +--------------------|------+------+------|
+     * | Ctrl |  `   | GUI  | Alt  |Lower |Space |  |Space |  |Space |Raise | Left | Down |  Up  |Right |
+     * `-----------------------------------------'  `------'  `-----------------------------------------'
+     */
+    [_QWERTY] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/ein_60/keymaps/ledtest/keymap.c:44]:
```suggestion
    [_QWERTY] = LAYOUT(
```

File: keyboards/ein_60/keymaps/ledtest/keymap.c [hunk 21]
```diff
@@ -0,0 +1,208 @@
...
+     * | Esc  |   !  |   @  |   #  |   $  |   %  |  |      |  |   ^  |   &  |   *  |   (  |   )  | Del  |
+     * |------+------+------+------+------+------|  `------'  |------+------+------+------+------+------|
+     * |  F1  |  F2  |  F3  |  F4  |  F5  |  F6  |            |      |   _  |   +  |   {  |   }  |Enter |
+     * |------+------+------+------+------+------|            |------+------+------+------+------+------|
+     * |  F7  |  F8  |  F9  |  F10 |  F11 |  F12 |            |      |      |      | Mute |      |  |   |
+     * |------+------+------|--------------------+  ,------.  +--------------------|------+------+------|
+     * |      |      |      |      |Lower | Bksp |  |      |  | Bksp |Raise | Next | Vol- | Vol+ | Play |
+     * `-----------------------------------------'  `------'  `-----------------------------------------'
+     */
+    [_LOWER] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/ein_60/keymaps/ledtest/keymap.c:61]:
```suggestion
    [_LOWER] = LAYOUT(
```

File: keyboards/ein_60/keymaps/ledtest/keymap.c [hunk 22]
```diff
@@ -0,0 +1,208 @@
...
+     * | Esc  |   1  |   2  |   3  |   4  |   5  |  |      |  |   6  |   7  |   8  |   9  |   0  | Del  |
+     * |------+------+------+------+------+------|  `------'  |------+------+------+------+------+------|
+     * |      |   4  |   5  |   6  |   +  |  F5  |            |  F6  |   -  |   =  |   [  |   ]  |Enter |
+     * |------+------+------+------+------+------|            |------+------+------+------+------+------|
+     * |Enter |   7  |   8  |   9  |   -  |  F11 |            |  F12 |ISO # |ISO / | Mute |      |   \  |
+     * |------+------+------|--------------------+  ,------.  +--------------------|------+------+------|
+     * |      |   ,  |   0  |  .   |Lower | Bksp |  |      |  | Bksp |Raise | Next | Vol- | Vol+ | Play |
+     * `-----------------------------------------'  `------'  `-----------------------------------------'
+     */
+    [_RAISE] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/ein_60/keymaps/ledtest/keymap.c:78]:
```suggestion
    [_RAISE] = LAYOUT(
```

File: keyboards/ein_60/keymaps/ledtest/keymap.c [hunk 23]
```diff
@@ -0,0 +1,208 @@
...
+     * |      |      |      |      |      |      |  |      |  |RGBRST|RGBMOD|RGBVAI|RGBSAI|RGBHUI|      |
+     * |------+------+------+------+------+------|  `------'  |------+------+------+------+------+------|
+     * |      |      |      |      |      |      |            |      |RGBRMO|RGBVAD|RGBSAD|RGBHUD|RGBTOG|
+     * |------+------+------+------+------+------|            |------+------+------+------+------+------|
+     * |      |      |      |      |      |      |            |      |      |      |      |      |BLSTEP|
+     * |------+------+------|--------------------+  ,------.  +--------------------|------+------+------|
+     * |      |      |      |      |      |      |  |      |  |      |      |      |      |      | RESET|
+     * `-----------------------------------------'  `------'  `-----------------------------------------'
+     */
+    [_ADJUST] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/ein_60/keymaps/ledtest/keymap.c:95]:
```suggestion
    [_ADJUST] = LAYOUT(
```

File: keyboards/ein_60/keymaps/ledtest/keymap.c [hunk 24]
```diff
@@ -0,0 +1,208 @@
...
+     * |      |      |      |      |      |      |  |      |  |      |      |  Up  |      |      |      |
+     * |------+------+------+------+------+------|  `------'  |------+------+------+------+------+------|
+     * |      |      |      |      |      |      |            |      | Left | Down |Right |      |      |
+     * |------+------+------+------+------+------|            |------+------+------+------+------+------|
+     * | Caps |      |      |      |      |      |            |      |      |      |      |      |      |
+     * |------+------+------|--------------------+  ,------.  +--------------------|------+------+------|
+     * |      |      |      |      |      |      |  |      |  |      |      |      |      |      |      |
+     * `-----------------------------------------'  `------'  `-----------------------------------------'
+     */
+    [_FUNCTION] = LAYOUT( \
```
Comment (fauxpark) [inline keyboards/ein_60/keymaps/ledtest/keymap.c:112]:
```suggestion
    [_FUNCTION] = LAYOUT(
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/ein_60/keymaps/ledtest/keymap.c [hunk 25]
```diff
@@ -0,0 +1,208 @@
...
+        }
+    } else if (index == 1) { /* Second encoder */
+        if (clockwise) {
+            tap_code(KC_DOWN);
+        } else {
+            tap_code(KC_UP);
+        }
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/ein_60/keymaps/ledtest/keymap.c:194]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by Klackygears** [comment]
Hey @drashna , regarding moving the oled stuff, I like that! Would it be easier to modify if its in a seperate file?

**Comment by drashna** [comment]
> Hey @drashna , regarding moving the oled stuff, I like that! Would it be easier to modify if its in a seperate file?

Nope.  I forgot to add a "weak" to it, though (fixed).  But the point for it being there is that it's included automatically with the keyboard, but can be replaced by specifying it in the keymap (or elsewhere).  This allows it to work if a user doesn't specify it, such as when compiling from the configurator.

**Comment by Klackygears** [comment]
> > Hey @drashna , regarding moving the oled stuff, I like that! Would it be easier to modify if its in a seperate file?
> 
> Nope. I forgot to add a "weak" to it, though (fixed). But the point for it being there is that it's included automatically with the keyboard, but can be replaced by specifying it in the keymap (or elsewhere). This allows it to work if a user doesn't specify it, such as when compiling from the configurator.

Oh! That makes sense! That's smart!

**Comment by Klackygears** [comment]
Hey @drashna I have a question. After making the changes to the oled, it's not playing nice with the layer indicators

```Compiling: keyboards/ein_60/ein_60.c                                                               keyboards/ein_60/ein_60.c: In function 'oled_task_user':
keyboards/ein_60/ein_60.c:64:14: error: '_QWERTY' undeclared (first use in this function)
         case _QWERTY:
              ^~~~~~~
keyboards/ein_60/ein_60.c:64:14: note: each undeclared identifier is reported only once for each function it appears in
keyboards/ein_60/ein_60.c:67:14: error: '_LOWER' undeclared (first use in this function); did you mean 'X_POWER'?
         case _LOWER:
              ^~~~~~
              X_POWER
keyboards/ein_60/ein_60.c:70:14: error: '_RAISE' undeclared (first use in this function); did you mean 'X_PAUSE'?
         case _RAISE:
              ^~~~~~
              X_PAUSE
keyboards/ein_60/ein_60.c:73:14: error: '_ADJUST' undeclared (first use in this function)
         case _ADJUST:
              ^~~~~~~
keyboards/ein_60/ein_60.c:76:14: error: '_FUNCTION' undeclared (first use in this function); did you mean 'QK_FUNCTION'?
         case _FUNCTION:
              ^~~~~~~~~
              QK_FUNCTION
 [ERRORS]
 |```
 
 I'm not sure how to correct this. What would be the best way?

**Comment by Klackygears** [comment]
I reverted the changes to the OLED because any time a layer was called, it caused errors especially in the circumstance that the layers were named differently that the default.

**Comment by drashna** [comment]
> Hey @drashna I have a question. After making the changes to the oled, it's not playing nice with the layer indicators
> 
>  
>  I'm not sure how to correct this. What would be the best way?

Basically just don't use the names.  Use numbers.  It's not as elegant, but ....  I would prefer it be there.

Probably can keep the code that is in the default keymap too.  That way, at least, it will work on the configurator.

**Comment by Klackygears** [comment]
> > Hey @drashna I have a question. After making the changes to the oled, it's not playing nice with the layer indicators
> > I'm not sure how to correct this. What would be the best way?
> 
> Basically just don't use the names. Use numbers. It's not as elegant, but .... I would prefer it be there.
> 
> Probably can keep the code that is in the default keymap too. That way, at least, it will work on the configurator.

So, something like what is in there now?

File: keyboards/ein_60/config.h [hunk 26]
```diff
@@ -0,0 +1,98 @@
...
+
+#define DIODE_DIRECTION COL2ROW
+
+#ifdef ENCODER_ENABLE
+#    define ENCODERS_PAD_A { B0, C5 }
+#    define ENCODERS_PAD_B { E3, C4 }
+#    define ENCODER_RESOLUTIONS { 4, 4 }
+#endif
+
+#ifdef OLED_DRIVER_ENABLE
```
Comment (drashna) [inline keyboards/ein_60/config.h:44]:
```suggestion
#ifdef OLED_ENABLE
```

File: keyboards/ein_60/keymaps/klackygears/keymap.c [hunk 27]
```diff
@@ -0,0 +1,176 @@
...
+      _______, _________________LYOUT_____________________,    _______,  _________________RGB_1_____________________, RGB_TOG,
+      _______, MU_TOG, _______, _______, _______,    CK_UP,              _________________RGB_2_____________________, _______,
+      RGB_MOD, MU_MOD,  MU_TOG,  CK_TOGG,  CK_RST, CK_DOWN,              _________________MEDIA_____________________, _______,
+      _______, _______, _______, _______, _______, _______,    _______,  _______, _______, _______, _______, _______, _______
+    )
+
+};
+
+
+#ifdef OLED_DRIVER_ENABLE
```
Comment (drashna) [inline keyboards/ein_60/keymaps/klackygears/keymap.c:90]:
```suggestion
#ifdef OLED_ENABLE
```

**Review by drashna — APPROVED** [review]
Yup!

**Comment by Klackygears** [comment]
> Yup!

Awesome! Thanks!

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Comment by Klackygears** [comment]
Thanks @drashna and @fauxpark !!!

**Pull request merged**